src/index.ts

```diff
@@ -1,22 +1,48 @@
 import axios from 'axios';
+import Ain from '@ainblockchain/ain-js';
 import Assets from './assets';
-import { ENDPOINT } from './constants';
+import Discord from './discord';
+import { AINFT_SERVER_ENDPOINT, BLOCKCHAIN_CHAINID, BLOCKCHAIN_ENDPOINT } from './constants';
+import { Account } from './types';
+
 export default class AinftJs {
   private baseUrl: string;
+  public accessAccount: Account;
+  public signature: string;
+  public signatureData: any;
   public assets: Assets;
+  public discord: Discord;
+  public ain: Ain;
 
-  constructor(baseUrl = ENDPOINT) {
+  constructor(baseUrl = AINFT_SERVER_ENDPOINT, accessAccount: Account) {
     this.baseUrl = baseUrl;
-    // TODO(hyeonwoong): setup for authorization. e.g. signatureData.
+    this.accessAccount = accessAccount;
+
+    this.ain = new Ain(BLOCKCHAIN_ENDPOINT, BLOCKCHAIN_CHAINID);
+    this.signatureData = Date.now().toString();
+    this.signature = this.buildSignature(this.signatureData);
 
     this.assets = new Assets(this.baseUrl);
+    this.discord = new Discord(
+      this.baseUrl,
+      this.accessAccount.address,
+      this.signature,
+      this.signatureData
+    );
   }
 
   setBaseUrl(baseUrl: string) {
     this.baseUrl = baseUrl;
+    this.assets.setBaseUrl(baseUrl);
+    this.discord.setBaseUrl(baseUrl);
+  }
+
+  buildSignature(data: any) {
```

Comment:
just sign?

Reply:
Yes. just sign and get signature. The signature will be used for authorization that user has access to app.

Reply:
Oh, sorry about confusion. I meant the name of the function. What about `signData`?

Reply:
This function is for creating a signature. Since the signature is used for authentication, it would be convenient to replace all signature-related functions if the authentication method is changed in the future.
But `signData` feels too generic. If we need sign later, we can make a util function and use that function in `buildSignature` :).